### PR TITLE
feat : add aria labelled as alias for accessibilityLabelledBy

### PR DESCRIPTION
**Pull request by dakshbhardwaj**
## Summary
This adds the [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby)  prop to the components where it's used as requested on https://github.com/facebook/react-native/issues/34424,  equivalent [accessibilityLabelledBy](https://reactnative.dev/docs/accessibility#accessibilitylabelledby-android) 

## Changelog
[General] [Added] - Add aria-modal prop to basic component

## TestPlan

 - Enable talkback.
 - Open the RNTester app and navigate to the Api's tab
 - Go to the TextInput with aria-labelledby attribute section
 
Screenshot 2022-09-19 at 7 46 05 PM

### CODE REVIEW

File: Libraries/Components/View/View.js [hunk 1]
```diff
@@ -161,6 +164,7 @@ const View: React.AbstractComponent<
           accessibilityElementsHidden={
             ariaHidden ?? accessibilityElementsHidden
           }
+          accessibilityLabelledBy={_accessibilityLabelledBy}
```
Comment (necolas) [inline Libraries/Components/View/View.js:167]: L174 is overriding this and some of the other props. The restProps stuff is a bit of a mess right now. Please could you fix that in this PR by moving all the prop extraction (including for accessibility value, state, etc) up to the block that is around L47. Then "otherProps" should be spread into the native component before all the specific accessibility props too. This is the kind of bug that snapshot tests could be detecting

**Review by necolas — CHANGES_REQUESTED** [review]
Let's add unit test snapshots because a number of these new props look like they are being overridden by the "restProps".

This prop should also be added to TextInput, Image, and Text. If there's no support for the underlying prop in RN, just adding the types will be enough.

File: Libraries/Components/View/ViewPropTypes.js [hunk 2]
```diff
@@ -542,6 +542,13 @@ export type ViewProps = $ReadOnly<{|
    */
   'aria-hidden'?: ?boolean,
 
+  /**
+   * It reperesents the nativeID of the associated label text. When the assistive technology focuses on the component with this props, the text is read aloud.
+   *
+   * @platform android
+   */
+  'aria-labelledby'?: ?string | ?Array<string>,
```
Comment (necolas) [inline Libraries/Components/View/ViewPropTypes.js:550]: Array of string isn't the web type. It would be comma separated IDs in a single string

**Comment by cipolleschi** [comment]
Hi @dakshbhardwaj! Thank you for taking this. Could you please address @necolas feedback? :pray:

**Comment by dakshbhardwaj** [comment]
@necolas @cipolleschi I have pushed down the changes

File: Libraries/Components/View/View.js [hunk 3]
```diff
@@ -45,19 +45,25 @@ const View: React.AbstractComponent<
...
       'aria-expanded': ariaExpanded,
       'aria-selected': ariaSelected,
-      ...restProps
-    } = otherProps;
+      'aria-labelledby': ariaLabelledBy,
+      'aria-valuemax': ariaValueMax,
+      'aria-valuemin': ariaValueMin,
+      'aria-valuenow': ariaValueNow,
+      'aria-valuetext': ariaValueText,
+      ...otherProps
```
Comment (necolas) [inline Libraries/Components/View/View.js:61]: Please can we alpha-sort props so it's easier to see the aria props grouped together etc. Same for the props applied to the native component. I'm surprised this isn't a lint rule.

File: packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js [hunk 4]
```diff
@@ -1446,4 +1446,18 @@ exports.examples = [
...
     },
   },
+  {
+    title: 'TextInput with aria-labelledby attribute"',
+    render(): React.Element<typeof View> {
+      return (
+        <View>
+          <Text nativeID="testAriaLabelledBy">Phone Number</Text>
+          <TextInput
+            aria-labelledby={'testAriaLabelledBy'}
```
Comment (necolas) [inline packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js:1456]: Where is the support for this implemented in TextInput, Text, Image components?

**Review by necolas — APPROVED** [review]

**Comment by necolas** [comment]
ping @@cipolleschi

File: packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js (hunk 4 above)
Comment (gabrieldonadel) [reply packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js:1456]: @necolas I was checking this to update the docs and it seems that `Text` doesn't ever support `accessibilityLabelledBy`

Comment (necolas) [reply packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js:1456]: Oh good to know

Comment (gabrieldonadel) [reply packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js:1456]: I've just opened a PR removing it, https://github.com/facebook/react-native/pull/35327

**Pull request closed without merging**